private/bufpkg/bufimage/build_image.go

```diff
@@ -33,6 +34,14 @@ import (
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
+// This enables editions support in protocompile. Buf doesn't yet support editions, but rejecting
+// such files happens elsewhere (in github.com/bufbuilder/buf/private/pkg/protodescriptor). We
+// need the compiler's support enabled so that we can start implementing and testing editions
+// support in the CLI, before we enable that support for users.
+//
+//go:linkname allowEditionsInCompiler github.com/bufbuild/protocompile/internal.AllowEditions
+var allowEditionsInCompiler = true
```

Comment:
Can we just have a `protocompile/editions.AllowEditions()` global function that we can call below? protocompile is pre-v1.0 so we can remove this later

Reply:
Eh, I suppose 🤷. TBH, I was really hoping to avoid anything in protocompile's exported API about editions until that repo's support is also complete. This seemed like the nicest way to preserve that property.

Reply:
Having run into a similar situation, a public opt-in doesn't seem great. Once editions have landed, the flag needs to be kept forever, but doesn't serve a purpose anymore.

Reply:
> the flag needs to be kept forever

This isn't necessarily an issue. As Peter mentioned, protocompile hasn't yet reached v1.0, so we can change the API and remove elements if need be (and I can document it as temporary, too).

I'm trying to think of what we'd name it to make it clear that users should generally _not_ use it as its incomplete. `AllowEditions` seems too innocent... Maybe `AllowEditionsThoughIncomplete`? Or perhaps it just live in a `testing` package, so it's more clear that it's for testing until the implementation is complete?

Reply:
@bufdev, @timostamm, how about this? https://github.com/bufbuild/protocompile/pull/267

Reply:
Updated this PR to use new opt-in function.